src/components/ha-expansion-panel.ts

```diff
@@ -0,0 +1,141 @@
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  LitElement,
+  property,
+  TemplateResult,
+} from "lit-element";
+import { fireEvent } from "../common/dom/fire_event";
+import "./ha-svg-icon";
+import { mdiChevronDown } from "@mdi/js";
+import { classMap } from "lit-html/directives/class-map";
+
+@customElement("ha-expansion-panel")
+class HaExpansionPanel extends LitElement {
+  @property({ type: Boolean, reflect: true }) expanded = false;
+
+  @property({ type: Boolean, reflect: true }) outlined = false;
+
+  protected render(): TemplateResult {
+    return html`
+      <div class="summary" @click=${this._toggleContainer}>
+        <slot name="title"></slot>
+        <ha-svg-icon
+          .path=${mdiChevronDown}
+          class="summary-icon ${classMap({ expanded: this.expanded })}"
+        ></ha-svg-icon>
+      </div>
+      <div class="container ${classMap({ expanded: this.expanded })}">
+        <slot></slot>
+      </div>
+    `;
+  }
+
+  protected firstUpdated(changedProps) {
+    super.firstUpdated(changedProps);
+
+    const container: HTMLDivElement = this.shadowRoot!.querySelector(
+      ".container"
+    )! as HTMLDivElement;
+
+    if (this.expanded) {
+      setTimeout(() => {
+        const scrollHeight = container.scrollHeight;
+        const animationDuration = this.getAutoHeightDuration(scrollHeight);
+        container.style.height = `${scrollHeight}px`;
+        container.style.transitionDuration = `${animationDuration}ms`;
+      }, 0);
+    }
+  }
+
+  private _toggleContainer(): void {
+    const container: HTMLDivElement = this.shadowRoot!.querySelector(
+      ".container"
+    )! as HTMLDivElement;
+    this.expanded = !this.expanded;
+
+    if (this.expanded) {
+      const scrollHeight = container.scrollHeight;
+      const animationDuration = this.getAutoHeightDuration(scrollHeight);
```

Comment:
Why do we want a variable duration?

Reply:
because I think the animation should be longer if the container if higher. We should have consistent speed.

Reply:
I think the animation time should be consistent.

Reply:
If the height will be 100px we should animate for 100ms (example time), but if the height will be 1000px the animation should be 1s, so the speed will be consistent.
This is how Material UI is doing, but I don't have a strong opinion about that, so I can set it to a fixed value.

Reply:
Got a link to the material specs?

Reply:
@bramkragten sadly there isn't any. Ref: https://github.com/material-components/material-components-web/issues/5939
Some info found here: https://material.io/design/motion/speed.html#duration

I've based this component on https://material-ui.com/components/accordion/

_Note: Accordions are no longer documented in the Material Design guidelines, but Material-UI will continue to support them. It was formerly known as the "expansion panel"._

Reply:
material-ui <> material.io

I would keep this as a reference: <https://material.io/design/motion/speed.html#duration>
Meaning you should choose between the medium or large, but not dynamic.

Reply:
So 300ms for expand and 250 for collapse would be OK? Or can we set this to a single value, let's say 300ms. This will be the easiest solution (no calculation at all). 

I personally think the length of transition should depend on content height, but let's make things simpler in first run. We can adjust that later 🙂

Reply:
Ok done. I've set a fixed value of 300ms.